### DISCUSSION TITLE
How to close ws connection when user logout

### DISCUSSION THREAD

**Discussion by juan2lv** [enisdenjo/graphql-ws#374] [Q&A] [answered]
When user logout, I would like to close the websocket connection. 
And when user login, I would like to reconnect websocket with the new auth token. 
Previously subscriptions-transport-ws has a close() function. How can I do this with graphql-ws?

Thank you!

**Comment by enisdenjo** [comment, marked as answer]
I recommend running the client in [`lazy` mode](https://github.com/enisdenjo/graphql-ws/blob/master/docs/interfaces/client.ClientOptions.md#lazy) which automatically closes the connection after all subscriptions complete and opens a new one when new subscriptions are made. This way, you don't have to manage the client manually, simply subscribe and unsubscribe when necessary.

**Reply by juan2lv** [reply]
Thanks!
